docs/getting-started-guides/fedora/fedora_manual_config.md

```diff
@@ -1,185 +1,228 @@
+    <pre><tt># <b>atomic host upgrade</b>
+   # <b>systemctl reboot</b>
+    </tt></pre>
 
-**System Information:**
+   On a Fedora Server or Fedora Workstation system, type:
 
-Hosts:
+    <pre><tt># <b>dnf update -y</b>
+   # <b>systemctl reboot</b>
+    </tt></pre>
 
-```conf
-fed-master = 192.168.121.9
-fed-node = 192.168.121.65
-```
+    Once the system comes up, log in again and run the next steps.
 
-**Prepare the hosts:**
+2. Disable swap. Comment out (#) any swap areas from /etc/fstab. For example, a commented out swap line in /etc/fstab might appear as follows:
 
-* Install Kubernetes on all hosts - fed-{master,node}.  This will also pull in docker. Also install etcd on fed-master.  This guide has been tested with Kubernetes-0.18 and beyond.
-* Running on AWS EC2 with RHEL 7.2, you need to enable "extras" repository for yum by editing `/etc/yum.repos.d/redhat-rhui.repo` and changing the `enable=0` to `enable=1` for extras.
+    <pre><tt>
+    <b>#</b> /dev/mapper/fedora--atomic-swap swap  swap  defaults  0 0
+    </tt></pre>
 
-```shell
-dnf -y install kubernetes
-```
+3. Disable selinux. Change the SELINUX type in /etc/sysconfig/selinux to permissive, so it appears as follows:
 
-* Install etcd
+        SELINUX=permissive
 
-```shell
-dnf -y install etcd
-```
+4. Immediately disable swap and SELinux:
 
-* Add master and node to `/etc/hosts` on all machines (not needed if hostnames already in DNS). Make sure that communication works between fed-master and fed-node by using a utility such as ping.
+    <pre><tt>
+    # <b>swapoff -a</b>
+    # <b>setenforce 0</b>
+    </tt></pre>
 
-```shell
-echo "192.168.121.9    fed-master
-192.168.121.65    fed-node" >> /etc/hosts
-```
+5. Disable firewalld. On Fedora Server or Workstation only (firewalld is no on Fedora Atomic by default), type the following.
 
-* Edit `/etc/kubernetes/config` (which should be the same on all hosts) to set
-the name of the master server:
+    <pre><tt>
+    # <b>systemctl stop firewalld</b>
+    # <b>systemctl disable firewalld</b>
+    </tt></pre>
 
-```shell
-# Comma separated list of nodes in the etcd cluster
-KUBE_MASTER="--master=http://fed-master:8080"
-```
+**WARNING**: Although disabling security features, such as SELinux and Firewalld, are currently needed to get this procedure to work properly, disabling them is not good practice when using Kubernetes in production. For that reason, this procedure is best used for trying and learning about Kubernetes in non-production environments.
```

Comment:
This deserves additional information. A pointer to a more thorough explanation of how to run these services in production with SELinux and Firewalld is fairly important if you really want behaviors around those security services to change. I want to run SELinux, I just never really see great documentation that helps me be successful in running SELinux

Reply:
I wasn't thinking of using kubeadm to set up a production-ready Kubernetes system. I thought it was a way to try out Kubernetes in a demo type of environment. With firewalld, I know at one point it caused problems with the way that Kubernetes used port forwarding. I suppose I could just set up some iptables firewall rules instead. With SELinux, I could create an SELinux policy, but that might require some explanation. Also, the kubeadm approach doesn't address issues like setting up storage or DNS. Do you think I need to add these features before publishing? Or could they be skipped or added as enhancements later?